ntbs-service/Services/SpecimenService.cs

```diff
@@ -0,0 +1,65 @@
+using System.Data.SqlClient;
+using System.Threading.Tasks;
+using Dapper;
+using Microsoft.Extensions.Configuration;
+using System.Linq;
+using System.Collections.Generic;
+using ntbs_service.Models.Entities;
+
+namespace ntbs_service.Services
+{
+    public interface ISpecimenService
+    {
+        Task<IEnumerable<Specimen>> GetSpecimenDetailsAsync(int notificationId);
+    }
+
+    public class SpecimenService : ISpecimenService

```

Comment:
I wonder if we want to combine the two services and make it possible to ask for one or both things - it would avoid us creating two separate connections, which can be quite a costly resource to obtain!

In fact thinking about it more, having the connection be created on each call to this doesn't seem right. It works well for migration (which is called very rarely and once it's done it's done), but for a service that will get called again and again just to display data, we probably want something a bit more robust. It might be worth reading around best practices for this a bit, but at the very least we could try this: create and open the connection in the constructor, make the serivce disposable and close the connection on dispose

Reply:
https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/sql-server-connection-pooling?redirectedfrom=MSDN

dotnet is already using connection pool when using single connections